### PR TITLE
localsend: 1.11.1 -> 1.12.0, build from source 

### PR DESCRIPTION
**Pull request by linsui**
## Description of changes

Because why not? And the closure size is smaller (1.7G -> 372.8M). The update script can't update the linux package though.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by adamcstephens — CHANGES_REQUESTED** [review]
Please rename the resulting binary, `result/bin/localsend_app` -> `result/bin/localsend`

Did you attempt to build from source on Darwin as well, or not?

**Comment by linsui** [comment]
> Did you attempt to build from source on Darwin as well, or not?

No, currently it's not supported.

https://github.com/NixOS/nixpkgs/blob/d5f43438f9f269f8748d2f982592a400b43f096c/pkgs/build-support/flutter/default.nix#L14

**Comment by linsui** [comment]
@sikmir Do you still maintain localsend?

**Review by adamcstephens — APPROVED** [review]
Result of `nixpkgs-review pr 259901` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>localsend</li>
    <li>localsend.debug</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 259901` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>localsend</li>
    <li>localsend.debug</li>
  </ul>
</details>

**Comment by sikmir** [comment]
> @sikmir Do you still maintain localsend?

Yes, LGTM.

File: pkgs/applications/networking/localsend/default.nix
Comment (sikmir) [inline pkgs/applications/networking/localsend/default.nix:26]: How to update deps.json on next release?

Comment (linsui) [reply pkgs/applications/networking/localsend/default.nix:26]: Remove this line and the vendorHash, then build the package. This file will be generated.

**Comment by linsui** [comment]
@sikmir Good to merge? :)

**Comment by sikmir** [comment]
[1.12.0](https://github.com/localsend/localsend/releases/tag/v1.12.0) is out meanwhile

**Comment by linsui** [comment]
I don't know how but the buildFlutterApplication is broken currently. See https://github.com/NixOS/nixpkgs/pull/263723.

File: pkgs/applications/networking/localsend/default.nix
Comment (hacker1024) [inline pkgs/applications/networking/localsend/default.nix:39]: This should be put in a [package override](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/compilers/dart/package-overrides).

#263345 will change how this works, though. I suggest waiting until that goes through, and then a nicer solution can be made.

Comment (linsui) [reply pkgs/applications/networking/localsend/default.nix:39]: > This should be put in a [package override](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/compilers/dart/package-overrides).

Thanks! Is there any documents?

> https://github.com/NixOS/nixpkgs/pull/263345 will change how this works, though. I suggest waiting until that goes through, and then a nicer solution can be made.

I'd like to wait for it if it can be merged in a week. I don't want to wait several months.

Comment (linsui) [reply pkgs/applications/networking/localsend/default.nix]: I moved the hack to package override. It's still hacky.

**Comment by hacker1024** [comment]
This PR should not include #263723.

File: pkgs/applications/networking/localsend/default.nix
Comment (hacker1024) [inline pkgs/applications/networking/localsend/default.nix:32]: There's no need to be manually choosing the `.dev` output here: https://nixos.org/manual/nixpkgs/unstable/#sec-multiple-outputs-using-split-packages

`lib.getDev` should have been used as well, `.dev` can break if the package ever switches to a single output.

File: pkgs/build-support/flutter/default.nix
Comment (hacker1024) [inline pkgs/build-support/flutter/default.nix:1]: This change should not be in this PR.

File: pkgs/development/compilers/dart/package-overrides/system-tray/default.nix
Comment (hacker1024) [inline pkgs/development/compilers/dart/package-overrides/system-tray/default.nix:15]: Normally patching the code like this would be a good solution, but modifying the package cache is a bit hacky. Would something like `patchelf --add-rpath` or `--add-needed` not work? The other alternative is adding to LD_LIBRARY_PATH.

This will be obsoleted by #263345 anyway, so I'm not too hung up on which solution we use for now.

**Review by hacker1024 — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/dart/package-overrides/system-tray/default.nix
Comment (linsui) [reply pkgs/development/compilers/dart/package-overrides/system-tray/default.nix:15]: I was told that such patch is preferred. :) Could you add something like `cargoDepsCopy` to `buildFlutterApplication`?

File: pkgs/build-support/flutter/default.nix
Comment (linsui) [reply pkgs/build-support/flutter/default.nix:1]: I'll remove it after the fix is merged.

File: pkgs/applications/networking/localsend/default.nix
Comment (linsui) [reply pkgs/applications/networking/localsend/default.nix:32]: Weird. It didn't build without the .dev. Removed.

**Comment by linsui** [comment]
> Please rename the resulting binary, `result/bin/localsend_app` -> `result/bin/localsend`
> 
> Did you attempt to build from source on Darwin as well, or not?

Localsend has a cli. It seems it's not ready yet but maybe we want the keep the localsend binary name for it.

**Comment by linsui** [comment]
This MR is still blocked by https://github.com/NixOS/nixpkgs/pull/263939.

**Comment by linsui** [comment]
@ofborg build localsend

File: pkgs/applications/networking/localsend/update.sh
Comment (adamcstephens) [inline pkgs/applications/networking/localsend/update.sh:2]: Why did you add a jq dependency here?

File: pkgs/applications/networking/localsend/update.sh
Comment (adamcstephens) [inline pkgs/applications/networking/localsend/update.sh:19]: Would you be willing to fix updating Linux instead of just removing it?

**Review by adamcstephens — COMMENTED** [review]
Result of `nixpkgs-review pr 259901` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>localsend</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 259901` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>localsend</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 259901` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>localsend</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 259901` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>localsend</li>
  </ul>
</details>

File: pkgs/applications/networking/localsend/update.sh
Comment (adamcstephens) [reply pkgs/applications/networking/localsend/update.sh:19]: If so, we should also add `passthru.updateScript = ./update.sh;`

File: pkgs/applications/networking/localsend/update.sh
Comment (linsui) [reply pkgs/applications/networking/localsend/update.sh:2]: It's used at line 8.

File: pkgs/applications/networking/localsend/update.sh
Comment (linsui) [reply pkgs/applications/networking/localsend/update.sh:19]: I don't know how to do that. I would like to have a try but not in this PR.

**Review by adamcstephens — APPROVED** [review]

**Comment by pbsds** [comment]
#259901

**Comment by linsui** [comment]
> Localsend has a cli. It seems it's not ready yet but maybe we want the keep the localsend binary name for it.

**Comment by pbsds** [comment]
Well, that goes against `mv $out/bin/localsend_app $out/bin/localsend` which was merged in this PR. Either the desktop application is moved back to `localsend_app`, or `mainProgram` is changed to point to `localsend`. The current state is that `nix run nixpkgs#localsend` does not works since it points at a binary that does not exist.

**Comment by linsui** [comment]
Oh, sorry, the line was added by mistake when I rebase the MR. I thought we should use localsend_app and leave localsend to the cli.

**Comment by linsui** [comment]
I opened https://github.com/NixOS/nixpkgs/pull/269253 instead.

**Pull request merged**
